pallets/parachain-staking/src/lib.rs

```diff
@@ -2048,6 +2054,14 @@ pub mod pallet {
 			let actual_weight = T::WeightInfo::delegator_bond_more(
 				<DelegationScheduledRequests<T>>::get(&candidate).len() as u32,
 			);
+
+			// If we have called delegator_bond_more for this deletagor before, the weight
+			// returned will be an over-estimate since the read was already performed on the first
+			// call and subsequent calls do not increase PoV size further.
+			if already_called_for_delegator {
+				actual_weight.set_proof_size(0);
```

Comment:
Seem's wrong to me as the only overlap is on the storage item DelegationScheduledRequests, it's better to apply x = 0 instead

Reply:
This is not going to work.
The way the benchmark is done is that it doesn't properly use the topDelegation storage properly and rely on the DelegationScheduledRequests argument passed, which is already 0 is most of the cases.

If we want to use this method of applying x = 0, then we need to review the weight (around 5kb for x = 0) of the benchmark, as this adds up to 1.5Mb with an empty storage (while in reality a full storage only goes to 150kb)